package.json

```diff
@@ -31,12 +35,16 @@
     "format": "prettier --write src 'src/**/*'",
     "lint": "eslint --fix --ext .js,.ts src",
     "build:js": "babel src --out-dir dist --extensions '.ts'",
+    "build:js-for-tests": "babel src --out-dir tests-js/src --extensions '.ts' --plugins transform-es2015-modules-commonjs",
+    "build:tests": "babel tests --out-dir tests-js/tests --extensions '.ts' --plugins transform-es2015-modules-commonjs",
     "build:types": "tsc",
-    "clean:js": "rimraf dist/*",
+    "clean:js": "rimraf dist",
+    "clean:tests": "rimraf tests-js",
     "clean:types": "rimraf index.d.ts index.d.ts.map",
     "build": "concurrently --names \"TYPES,BABEL,LINTS\" --prefix \"[{name}]\" --prefix-colors \"bgBlue.white.bold,bgYellow.black.bold,bgMagenta.white.bold\" \"npm run build:types\" \"npm run build:js\" \"npm run lint\"",
-    "clean": "concurrently --names \"TYPES,BABEL\" --prefix \"[{name}]\" --prefix-colors \"bgBlue.white.bold,bgYellow.black.bold\"  \"npm run clean:js\" \"npm run clean:types\"",
-    "pretest": "npm run build",
+    "clean": "concurrently --names \"TYPES,BABEL,TESTS\" --prefix \"[{name}]\" --prefix-colors \"bgBlue.white.bold,bgYellow.black.bold,bgGreen.white.bold\"  \"npm run clean:js\" \"npm run clean:types\" \"npm run clean:tests\"",
+    "pretest": "concurrently --names \"BABEL,TESTS\" --prefix \"[{name}]\" --prefix-colors \"bgYellow.black.bold,bgGreen.white.bold\" \"npm run build:js-for-tests\" \"npm run build:tests\"",
+    "test": "qunit 'tests-js/**/*.js'",
```

Comment:
Still TODO: implement `watch:tests`. Can work straightforwardly on top of the existing primitives.